src/client/mod.rs

```diff
@@ -303,6 +304,50 @@ impl KafkaClient {
         }
     }
 
+    // Side note: The below example MUST have a `main` function delcared to pull in the extern crate.
+
+    /// Creates a new instance of KafkaClient with SSL secured connections.
+    /// This returns a `Result<KafkaClient>` because all connections are
+    /// attempted at creation time. Errors thrown here will be SSL related.
+    /// Before being able to successfully use the new client, you'll have to
+    /// load metadata.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// extern crate openssl;
+    /// extern crate kafka;
+    /// use openssl::ssl::{Ssl, SslContext, SslStream, SslMethod, SSL_VERIFY_NONE};
+    /// use openssl::x509::X509FileType;
+    /// use kafka::client::KafkaClient;
+    /// use std::path::Path;
+    ///
+    /// fn main() {
+    ///     let (key, cert) = ("foo.key".to_string(), "bar.crt".to_string());
+    ///
+    ///      // OpenSSL offers a variety of complex configurations. Here is an example:
+    ///      let mut context = SslContext::new(SslMethod::Sslv23).unwrap();
+    ///      context.set_cipher_list("DEFAULT").unwrap();
+    ///      context.set_certificate_file(&cert, X509FileType::PEM).unwrap();
+    ///      context.set_private_key_file(&key, X509FileType::PEM).unwrap();
+    ///      context.set_verify(SSL_VERIFY_NONE, None);
+    ///      let ssl = Ssl::new(&context).unwrap();
+    ///
+    ///      let mut client = KafkaClient::new_secured(vec!("localhost:9092".to_owned()), ssl).unwrap();
+    ///      client.load_metadata_all().unwrap();
+    /// }
+    /// ```
+    ///
+    /// See also `KafkaClient::load_metadatata_all` and `KafkaClient::load_metadata`
+    pub fn new_secured(hosts: Vec<String>, ssl: Ssl) -> Result<KafkaClient> {
+        let mut client = KafkaClient::new(hosts);
+        for host in &client.config.hosts {
+            let conn = try!(client.conn_pool.get_conn(host));
+            try!(conn.ssl(ssl.clone()));
+        }
+        Ok(client)
+    }
```

Comment:
i think this is problematic for the following reasons:

1. i think it should be valid for some of the user provided bootstrap hosts (i.e. `client.config.hosts`) not to be available. if i'm not mistaken, this could would fail as soon as at least one is not reachable. 
2. further, i think we'll need to move the "enforce ssl comunication on a connection" directly into the connection pool. Upon initialization, `KafkaClient::fetch_metadata` contacts the bootstrap servers to learn about the whole cluster, thereby receiving new host names, to which kafka-rust will later establish connections to. For example - out of laziness - I often specify only one bootstrap server, while the whole cluster consists of multiple machines/hosts. In this scenario, the new code would establish secured communication only to the one i explicitly specified, but would continue speaking to the rest of the cluster in plain text.